Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -181,6 +171,73 @@ theorem prod_trunc {M : Type*} [CommMonoid M] {a b : ℕ} (f : Fin (a + b) → M
   rw [prod_univ_add, Fintype.prod_eq_one _ hf, mul_one]
   rfl
 
+section Interval
+
+variable {M : Type*} [CommMonoid M] {n : ℕ} {v : Fin (n + 1) → M}
+
+@[to_additive]
+theorem prod_Ioi_zero : ∏ i ∈ Ioi 0, v i = ∏ j : Fin n, v j.succ := by
+  rw [Ioi_zero_eq_map, Finset.prod_map, val_succEmb]
+
+@[to_additive]
+theorem prod_Ioi_succ (i : Fin n) : ∏ j ∈ Ioi i.succ, v j = ∏ j ∈ Ioi i, v j.succ := by
+  rw [Ioi_succ, Finset.prod_map, val_succEmb]
+
+@[to_additive (attr := simp)]
+theorem prod_Ici_zero : ∏ i ∈ Ici 0, v i = ∏ j : Fin (n + 1), v j := by
+  rw [Ici_eq_Icc, ← bot_eq_zero, Icc_bot_top]
+
+@[to_additive]
+theorem prod_Ici_succ (i : Fin n) :
+    ∏ j ∈ Ici i.succ, v j = ∏ j ∈ Ici i, v j.succ := by
+  rw [Ici_succ, Finset.prod_map, val_succEmb]
+
+@[to_additive (attr := simp)]
+theorem prod_Iio_zero : ∏ i ∈ Iio 0, v i = 1 := rfl
```

Comment:
And Iio 0 = empty would be good too. Can this be stated for all canonically ordered types?